src/Agent.Plugins/TestResultParser/Plugin/TestResultLogPlugin.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Agent.Plugins.Log.TestResultParser.Contracts;
+using Agent.Plugins.Log.TestResultParser.Plugin;
+using Agent.Plugins.TestResultParser.Plugin;
+using Agent.Sdk;
+using Pipelines = Microsoft.TeamFoundation.DistributedTask.Pipelines;
+
+namespace Agent.Plugins.Log
+{
+    public class TestResultLogPlugin : IAgentLogPlugin
+    {
+        /// <inheritdoc />
+        public string FriendlyName => "Test Result Log Parser";
+
+        /// <inheritdoc />
+        public async Task<bool> InitializeAsync(IAgentLogPluginContext context)
+        {
+            try
+            {
+                _logger = new TraceLogger(context);
+                _clientFactory = new ClientFactory(context.VssConnection);
+
+                PopulatePipelineConfig(context);
+
+                if (CheckForPluginDisable(context))
+                {
+                    return false; // disable the plugin
+                }
+
+                await InputDataParser.InitializeAsync(_clientFactory, _pipelineConfig, _logger);
+            }
+            catch (Exception ex)
+            {
+                _logger.Warning($"Unable to initialize {FriendlyName}");
+                context.Trace(ex.StackTrace);
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <inheritdoc />
+        public async Task ProcessLineAsync(IAgentLogPluginContext context, Pipelines.TaskStepDefinitionReference step, string line)
+        {
+            await InputDataParser.ProcessDataAsync(line);
+        }
+
+        /// <inheritdoc />
+        public async Task FinalizeAsync(IAgentLogPluginContext context)
+        {
+            await InputDataParser.CompleteAsync();
+        }
+
+        private bool CheckForPluginDisable(IAgentLogPluginContext context)
+        {
+            if (context.Variables.TryGetValue("test.disable.noconfig", out var disable) && !string.IsNullOrEmpty(disable.Value))
+            {
+                return true;
+            }
+
+            // check for PTR task or some other tasks to enable/disable
+            return context.Steps == null
+                   || context.Steps.Any(x => x.Id.Equals(new Guid("0B0F01ED-7DDE-43FF-9CBB-E48954DAF9B1")))
+                   || _pipelineConfig.BuildId == 0
+                   || !IsValidGithubBuild(context);
+        }
+
+        private bool IsValidGithubBuild(IAgentLogPluginContext context)
+        {
+            // disable in release pipeline
+            if (context.Variables.TryGetValue("release.releaseId", out var releaseIdValue)
+                && int.TryParse(releaseIdValue.Value, out var releaseId) && releaseId > 0)
+            {
+                return false;
+            }
+
+            // enable for github only
```

Comment:
this seems really weird, why only github?

Reply:
@TingluoHuang We wanted to start off with Github projects only (public projects).
The goal is to let customers aware of test experience and migrate to full reporting with Junit.

Parsing currently gets you into basic reporting along with test experience.
Analytics and full richer reporting available via Junit reporting.